static/js/realm_logo.js

```diff
@@ -13,29 +13,34 @@ exports.build_realm_logo_widget = function (upload_function, is_night) {
         return file_input_elem.expectOne();
     };
 
-    if (page_params.realm_logo_source === 'D') {
-        delete_button_elem.hide();
-    } else {
-        delete_button_elem.show();
+    if (page_params.is_admin) {
+        if (page_params.realm_logo_source === 'D') {
+            delete_button_elem.hide();
+        } else {
+            delete_button_elem.show();
+        }
     }
 
     const data = {night: JSON.stringify(is_night)};
```

Comment:
This can go inside the `if` block .. and then you can actually do early return and just have `if (!page_params.is_admin) { return;) rather than reindenting things?

Reply:
Done. Thanks.